clients/fides-js/src/components/tcf/InitialLayer.tsx

```diff
@@ -7,46 +7,67 @@ import {
   getIdsNotRepresentedInStacks,
 } from "../../lib/tcf/stacks";
 import InitialLayerAccordion from "./InitialLayerAccordion";
+import {
+  TCFPurposeConsentRecord,
+  TCFPurposeLegitimateInterestsRecord,
+} from "../../lib/tcf/types";
 
 const InitialLayer = ({ experience }: { experience: PrivacyExperience }) => {
-  const purposeIds = useMemo(
-    () =>
-      experience.tcf_purposes ? experience.tcf_purposes.map((p) => p.id) : [],
-    [experience.tcf_purposes]
-  );
+  const {
+    tcf_consent_purposes: consentPurposes = [],
+    tcf_legitimate_interests_purposes: legintPurposes = [],
+    tcf_special_features: experienceSpecialFeatures = [],
+  } = experience;
+
+  const uniquePurposeIds = useMemo(() => {
```

Comment:
we now have two fields: `tcf_consent_purposes` and `tcf_legitimate_interests_purposes` which can overlap! this means, if we only want one instance of a purpose, we need to dedupe